src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -114,4 +134,41 @@ public static String evalXpathExpression(XPathExpression expr,
         return new Pair<>(ExternalDataInstance.buildFromRemote(queryDatum.getDataId(), root, queryDatum.useCaseTemplate()), "");
     }
 
+    public void populateItemSetChoices(QueryPrompt queryPrompt) {
+        evaluationContext.setVariables(userAnswers);
```

Comment:
If you are going to change scoped values in an EvaluationContext you should always spawn a new context (`evaluationContext.spawnWithCleanLifecycle()`) before doing so, since you don't want to change the "root" context data for future requests.